src/avro_copy.cpp

```
@@ -0,0 +1,498 @@
+		idx_t length = datum_buffer.GetCapacity() - offset_in_datum_buffer;
+		if (value_size > length) {
+			//! This value is too big to fit into the remaining portion of the buffer
+			idx_t new_capacity = datum_buffer.GetCapacity();
+			new_capacity += value_size;
+			datum_buffer.ResizeAndCopy(NextPowerOfTwo(new_capacity));
+		}
+		avro_writer_memory_set_dest_with_offset(global_state.datum_writer, (const char *)datum_buffer.GetData(),
+		                                        datum_buffer.GetCapacity(), offset_in_datum_buffer);
+
+		while (avro_file_writer_append_value(global_state.file_writer, &local_state.value)) {
+			auto current_capacity = datum_buffer.GetCapacity();
+			datum_buffer.ResizeAndCopy(NextPowerOfTwo(current_capacity * 2));
+			avro_writer_memory_set_dest_with_offset(global_state.datum_writer, (const char *)datum_buffer.GetData(),
+			                                        datum_buffer.GetCapacity(), offset_in_datum_buffer);
+		}
+
+		offset_in_datum_buffer = avro_writer_tell(global_state.datum_writer);
+		avro_value_reset(&local_state.value);
+	}
+
+	auto &buffer = global_state.memory_buffer;
+	auto expected_size = avro_writer_tell(global_state.datum_writer);
+	expected_size += 16 + 4;
+	if (expected_size > buffer.GetCapacity()) {
+		//! Resize the buffer in advance, to prevent any need for resizing below
+		buffer.Resize(NextPowerOfTwo(expected_size));
+		avro_writer_memory_set_dest(global_state.writer, (const char *)buffer.GetData(), buffer.GetCapacity());
+	}
+
+	//! Flush the contents to the buffer, if it fails, resize the buffer and try again
+	while (avro_file_writer_flush(global_state.file_writer)) {
+		auto current_capacity = buffer.GetCapacity();
+		buffer.Resize(NextPowerOfTwo(current_capacity * 2));
+		avro_writer_memory_set_dest(global_state.writer, (const char *)buffer.GetData(), buffer.GetCapacity());
+	}
+
+	auto written_bytes = avro_writer_tell(global_state.writer);
+	global_state.WriteData(buffer.GetData(), written_bytes);
```

maybe write data can return a return code? 
Then if the write fails we can return a nice error message here? Seems like `handle->Write((void *)data, size);` in `WriteAvroGlobalState::WriteData` does this